tests/integration_tests/base_tests.py

```diff
@@ -106,6 +106,9 @@ class SupersetTestCase(TestCase):
 
     maxDiff = -1
 
+    def tearDown(self):
```

Comment:
The whole login/logout logic is convoluted given how our tests are defined. Ideally there would be an `auth_user` context manager and an associated `pytest` fixture which would be used to authenticate a user. Sadly this isn't overly viable with the `pytest`/`unittest` mix we have and so, although overkill—given that not every tests uses the API—this seemed like the best mechanism (given that all `unittest` classes are derived from `SupersetTestCase`) for ensuring that the tests are idempotent.